### ISSUE TITLE
`ResponseUITestTrait` | `WebUITestTrait`: Test domain used for urls and App domains during testing is not consistent

### ISSUE THREAD

**Issue by sevidmusic** [sevidmusic/DarlingDataManagementSystem#372]
# `TLDR;`
Both the [`ResponseUITestTrait`](https://github.com/sevidmusic/roady/blob/roady/Tests/Unit/interfaces/component/UserInterface/TestTraits/ResponseUITestTrait.php)  and the [`WebUITestTrait`](https://github.com/sevidmusic/roady/blob/roady/Tests/Unit/interfaces/component/UserInterface/TestTraits/WebUITestTrait.php) are not using consistent domains or urls during testing.

Define a new `protected` `static` property named `$testDomain` in the [`ResponseUITestTrait`](https://github.com/sevidmusic/roady/blob/roady/Tests/Unit/interfaces/component/UserInterface/TestTraits/ResponseUITestTrait.php) and use it to assign domains or urls during testing, for example as the domain to build an App for, or as the url assigned to a `Request`.

# Details:

`ResponseUITestTrait`: The test domains and urls used during testing are not set to a consistent value. 

This also effects the `WebUITestTrait`, and the issue was noticed while addressing issue #371.

For example the `getRequest()` method defined by the [`WebUITestTrait`](https://github.com/sevidmusic/roady/blob/roady/Tests/Unit/interfaces/component/UserInterface/TestTraits/WebUITestTrait.php), which is used to get a `Request` instance for testing is:
```
public static function getRequest(): RequestInterface
    {
        $request = new Request(
            new Storable(
                'ResponseUICurrentRequest' . strval(rand(0, 999)),
                self::expectedAppLocation(),
                self::getTestComponentContainer()
            ),
            new Switchable()
        );
        $request->import(['url' => './?request=' . self::$requestedStylesheetNameA . '&request=' . self::$requestedStylesheetNameB]);
        return $request;
    }

    /**
```

Whereas the `getRequest()` method defined by the `ResponseUITestTrait` is:
```
    public static function getRequest(): RequestInterface
    {
        return new CoreRequest(
            new CoreStorable(
                'ResponseUICurrentRequest' . strval(rand(0, 999)),
                self::expectedAppLocation(),
                self::getTestComponentContainer()
            ),
            new CoreSwitchable()
        );
    }
```
And the `getIndependantRequest()` method defined by the `ResponseUITEstTrait` is:
```
    public static function getIndependantTestRequest(): RequestInterface
    {
        $request =  new CoreRequest(
            new CoreStorable(
                'ResponseUICurrentRequest' . strval(rand(0, 999)),
                'TestUrls',
                self::getTestComponentContainer()
            ),
            new CoreSwitchable()
        );
        return $request;
    }
```

All of these methods are used to get a Request instance for testing in different contexts, but all assign a different url (or dont assing one at all), and none of them use a valid domain as part of the url.

The purpose of the `ResponseUITestTrait`'s `getRequest()` method is to return a `Request` instance that can be assigned to the `Router` used during testing.  

The `getIndependantTestRequest()` is intended to be used soley by `expectedAppLocation()` to provide a `Request` instance to `App::deriveAppLocationFromRequest()` to avoid the infinite recursive call loop that would occur if `getRequest()` was used.

After `$testDomain` property is defined, refactor the `ResponseUITestTrait`'s `getRequest()` method to be:

```
    public static function getRequest(): RequestInterface
    {
        $request = new CoreRequest(
            new CoreStorable(
                'ResponseUICurrentRequest' . strval(rand(0, 999)),
                self::expectedAppLocation(),
                self::getTestComponentContainer()
            ),
            new CoreSwitchable()
        );
        $request->import(['url' => self::$testDomain]);
        return $request;
    }
```
And the `ResponseUITestTrait`'s `getIndependantTestRequest()` method to be:

```
    public static function getIndependantTestRequest(): RequestInterface
    {
        $request =  new CoreRequest(
            new CoreStorable(
                'ResponseUIIndependantRequest' . strval(rand(0, 999)),
                'ResponseUIIndependantRequests',
                self::getTestComponentContainer()
            ),
            new CoreSwitchable()
        );
        $request->import(['url' => self::$testDomain]);
        return $request;
    }
```

And finally, the `WebUITestTrait`'s  `getRequest()` method to be:

```
 public static function getRequest(): RequestInterface
    {
        $request = new Request(
            new Storable(
                self::getUniqueName('Request'),
                self::expectedAppLocation(),
                self::getTestComponentContainer()
            ),
            new Switchable()
        );
        $request->import(['url' => self::$testDomain . '/?request=' . self::$requestedStylesheetNameA . '&request=' . self::$requestedStylesheetNameB]);
        return $request;
    }
```

**Comment by sevidmusic** [comment]
Resolved in commit b9b9a87346ef5ca591306ca74878edee553a67bd

**Comment by sevidmusic** [comment]
Resolved in commit b9b9a87346ef5ca591306ca74878edee553a67bd
